config/backend-config/backend-config.go

```diff
@@ -51,6 +51,7 @@ type workspaceConfig interface {
 	GetWorkspaceIDForWriteKey(string) string
 	GetWorkspaceIDForSourceID(string) string
 	GetWorkspaceLibrariesForWorkspaceID(string) LibrariesT
+	GetDataRetentionSettingsForWorkspaceID(string) DataRetention
```

Comment:
We can make our methods a bit more readable and idiomatic by:
* remove `Get`
* `ForWorkspaceID` we can assume workspaceID is the default means of lookup and only specify methods that don't follow the standard, example`ForWriteKey`.  The name param can provide additional documentation here.

```suggestion
	DataRetentionSettings(workspaceID string) DataRetention
```

For consistency we should rename the other methods as well.

Reply:
I am also wondering whether we need a new method/map for  `DataRetentionSettings` or we can just use `ConfigT`.

On the one hand, using specific data structs reduces the coupling for the components that depend on the config; on the other hand, the config becomes polluted with all the specific components we want to configure.

Without getting outside of the scope of this PR, I would suggest merging:
```go
	wc.workspaceIDToLibrariesMap = workspaceIDToLibrariesMap
	wc.workspaceIDToDataRetentionMap = workspaceIDToDataRetentionMap
 ```
 into a
 ```go
  map[string]ConfigT
  ```
  
  While maintaining the existing method signatures.

Reply:
Addressed first comment. Second one, we can do all of them together later.